spec/requests/api/v1/targets/match_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+
+describe 'Match Targets', type: :request do
+  subject { post api_v1_targets_path, params:, headers:, as: :json }
+
+  let!(:user) { create(:user) }
+  let!(:another_user) { create(:user) }
+  let(:headers) { another_user.create_new_auth_token }
+  let(:topic) { create(:topic) }
+  let!(:target) { create(:target, user:, location: "POINT (-63.59386082773612 -5.199173507365742)", topic:) }
+  let(:title) { Faker::Lorem.word }
+  let(:radius) { Faker::Number.number(digits: 3) }
+  let(:latitude) { Faker::Address.latitude }
+  let(:longitude) { Faker::Address.longitude }
+  let(:params) do
+    {
+      target: {
+        title:,
+        radius:,
+        topic_id: topic.id
+      },
+      latitude: -63.59386082773612,
+      longitude: -5.199173507365742
+    }
+  end
+
+  context 'when success' do
+    it 'should return a sucessfull response' do
+      subject
+      expect(response).to have_http_status(:created)
+    end
+
+    it 'should notify the owner of the created target' do
+      expect { subject }.to change(ActionMailer::Base.deliveries, :count).by(1)
+    end
+
+    it 'should create the conversation between the users' do
+      expect { subject }.to change(Conversation, :count).by(1)
+    end
+  end
+
+  context 'when fails' do
```

Comment:
Could you add some other examples?
 - Same topic but not in range
 - In range but different topic
 - Same topic, in range but from the same user